revup/amend.py

```diff
@@ -16,8 +16,14 @@
 
 RE_TOPIC_WITH_MODIFIERS = re.compile(r"(?P<topic>[a-zA-Z\-_0-9]+)(?P<modifiers>[\^~]+[0-9]*)?")
 
+CLEANUP_SCISSOR_LINE = r"------------------------ >8 ------------------------"
+CLEANUP_SCISSOR_COMMENT = """Do not modify or remove the line above.
+Everything below it will be ignored."""
+CLEANUP_STRIP_COMMENT = """Please enter the commit message for your changes. Lines starting
+with '#' will be ignored, and an empty message aborts the amend."""
 
-def invoke_editor_for_commit_msg(
+
+async def invoke_editor_for_commit_msg(
```

Comment:
Weird side question - are we worried at all about invoking multiple git commands in async threads that write to the repo state?  I'm assuming this is fine but idk if we have any convention or whatever to protect against that

Reply:
here the new commands don't actually change state, they just read state

in general though git has internal file based locking so parallel git commands can't result in invalid state. in revup we don't generally use parallel commands that touch the same state though

Reply:
That's cool, I'm probably just being a bit paranoid because I recently ran a garbage collect from one worktree during a rebase in another worktree that very much broke things (not in a sense that actual git state was broken I guess, but missing objects and unhappy reflog and such)